### PR TITLE
Split `Excel_Format` into 3 constructors

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

Splitting the `Excel_Format` into 3 constructors.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] Ensure that the `Data.read` correctly allows to pick any of the constructors.
- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by radeusgd** [comment]
image

**Comment by radeusgd** [comment]
image

I'm trying to test the dropdowns, but even on an engine build from `develop` + old IDE build, I seem to be getting problems - arguments of the inserted format - be it `Delimited` or `Plain_Text` do not show up. Thus I'm unable to verify if my PR breaks something or not.

**Comment by radeusgd** [comment]
On a nightly from yesterday, e.g. on Delimited only the delimiter shows up and not all the others. Is that intentional? cc: @AdRiley @jdunkerley 
image
(this is a screenshot from a develop engine)

File: distribution/lib/Standard/Database/0.0.0-dev/src/Connection/SQLite_Format.enso [hunk 1]
```diff
@@ -36,6 +37,10 @@ type SQLite_Format
             ".sqlite" -> SQLite_Format.For_File
             _ -> Nothing
 
+    ## PRIVATE
+    get_dropdown_options : Vector Option
+    get_dropdown_options = [Option "SQLite Database File" (Meta.get_qualified_type_name SQLite_Format)+".For_File"]
```
Comment (jdunkerley) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Connection/SQLite_Format.enso]:
```suggestion
    get_dropdown_options = [Option "SQLite" (Meta.get_qualified_type_name SQLite_Format)+".For_File"]
```

**Review by jdunkerley — APPROVED** [review]
Just for clarity I'd move `Excel_Section` to `Internal` but looks great.

**Comment by jdunkerley** [comment]
> On a nightly from yesterday, e.g. on Delimited only the delimiter shows up and not all the others. Is that intentional? cc: @AdRiley @jdunkerley 
> 
> image
> 
> (this is a screenshot from a develop engine)

Yes we finally have parameters following the display logic. So only show if Always or if modified. 

When you select you see them.

**Review by AdRiley — APPROVED** [review]
I like this change :) The headers option is so much better now.

File: distribution/lib/Standard/Database/0.0.0-dev/src/Connection/SQLite_Format.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Database/0.0.0-dev/src/Connection/SQLite_Format.enso]: ok

**Comment by radeusgd** [comment]
> Yes we finally have parameters following the display logic. So only show if Always or if modified.
> 
> When you select you see them.

At thanks, indeed now I can see it all works fine. I was worried I broke something but apparently it's a new feature 😅

**Pull request merged**